app/src/main/kotlin/at/bitfire/davdroid/sync/worker/BaseSyncWorker.kt

```diff
@@ -142,16 +141,15 @@ abstract class BaseSyncWorker(
     suspend fun doSyncWork(account: Account, dataType: SyncDataType): Result {
         logger.info("Running ${javaClass.name}: account=$account, dataType=$dataType")
 
-        // pass possibly supplied flags to the selected syncer
-        val extrasList = mutableListOf<String>()
-        when (inputData.getInt(INPUT_RESYNC, NO_RESYNC)) {
-            RESYNC ->      extrasList.add(Syncer.SYNC_EXTRAS_RESYNC)
-            FULL_RESYNC -> extrasList.add(Syncer.SYNC_EXTRAS_FULL_RESYNC)
+        // pass supplied parameters to the selected syncer
+        val resyncType: ResyncType? = when (inputData.getInt(INPUT_RESYNC, NO_RESYNC)) {
+            RESYNC_ENTRIES -> ResyncType.RESYNC_ENTRIES
+            RESYNC_LIST -> ResyncType.RESYNC_LIST
+            else -> null
         }
-        if (inputData.getBoolean(INPUT_UPLOAD, false))
-            // Comes in through SyncAdapterService and is used only by ContactsSyncManager for an Android 7 workaround.
-            extrasList.add(ContentResolver.SYNC_EXTRAS_UPLOAD)
-        val extras = extrasList.toTypedArray()
+
+        // Comes in through SyncAdapterService and is used only by ContactsSyncManager for an Android 7 workaround.
+        val syncFrameworkUpload = inputData.getBoolean(INPUT_UPLOAD, false)
 
         // We still use the sync adapter framework's SyncResult to pass the sync results, but this
         // is only for legacy reasons and can be replaced by our own result class in the future.
```

Comment:
We can remove this comment, since we are using our own `SyncResult` implementation now.